servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -592,7 +615,9 @@ private void processWorkerAddedBlocks(MasterWorkerInfo workerInfo,
    * @return the lost blocks in Tachyon Storage
    */
   public Set<Long> getLostBlocks() {
-    return Collections.unmodifiableSet(mLostBlocks);
+    synchronized (mLostBlocks) {
+      return ImmutableSet.copyOf(mLostBlocks);
+    }
   }
```

Comment:
Wrapping the set in a Collections.unmodifiable thinghie is not enough. What if mLostBlocks is modified while another thread is going through its unmodifiable _view_?